### PR TITLE
Fixes #1294: Configure GithubProfileScraper with Post and BaseScraper

### LINKED ISSUE #1294: Configure GithubProfileScraper with Post and BaseScraper
- Issue type: Feature request
- Parent Issue: Issue #1129

Tasks:

1) Configure `GithubProfileScraper.java` with `BaseScraper.java`
2) Create Post Object for `GithubProfileScraper.java`

### PR DESCRIPTION
**Pull request by vibhcool**
PR fixes:

1) Issue #1294 
2) Update the test

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by vibhcool** [comment]
open for review. tested and codacy issues fixed
other codacy issues can't be fixed
@Achint08 @djmgit @hemantjadon @kavithaenair @SKrPl @singhpratyush

File: src/org/loklak/api/search/QuoraProfileScraper.java [hunk 1]
```diff
@@ -53,18 +53,20 @@ public QuoraProfileScraper() {
         super();
```
Comment (djmgit) [inline src/org/loklak/api/search/QuoraProfileScraper.java]: Can the changes to QuoraProfileScraper be moved to a separate PR as this PR as the name suggests deals with configuring GithubProfileScraper.

Comment (vibhcool) [inline src/org/loklak/api/search/QuoraProfileScraper.java]: I have made no changes in QuoraProfileScraper, these changes are to prevent any error, if it occurs due to configurations in GithubProfileScraper. Before GithubProfileScraper, BaseScraper was only used in QuoraProfileScraper and it is new, so to be make it more generic some changes are necessary.

File: src/org/loklak/harvester/BaseScraper.java [hunk 2]
```diff
@@ -107,11 +114,16 @@ public Post getDataFromConnection(String url, String type) throws IOException {
...
     }
-/*
+
     public Post getDataFromConnection(String url) throws IOException {
         return getDataFromConnection(url, "all");
     }
-*/
+
+    public Post getDataFromConnection() throws IOException {
+        String url = this.prepareSearchUrl("all");
```
Comment (singhpratyush) [inline src/org/loklak/harvester/BaseScraper.java]: Why hard coding `"all"` in `BaseScraper`? Why is it needed here?

File: test/org/loklak/api/search/GithubProfileScraperTest.java [hunk 3]
```diff
@@ -1,45 +1,81 @@
...
+            String specialLink = "http://djmgit.github.io";
+            String userId = "16368427";
+
+            Post fetchedProfile = githubScraper.scrapeGithub(profile, br);
+
+            assertEquals(fetchedProfile.getString("user_name"), userName);
+            assertEquals(fetchedProfile.getString("full_name"), fullName);
+            assertEquals(fetchedProfile.getString("special_link"), specialLink);
+            assertEquals(fetchedProfile.getString("user_id"), userId);
+        } catch (IOException e) {
```
Comment (singhpratyush) [inline test/org/loklak/api/search/GithubProfileScraperTest.java]: So the test will pass in case of `IOException`. Is this desirable?

Comment (vibhcool) [inline test/org/loklak/api/search/GithubProfileScraperTest.java]: `IOException` is needed to be handled as ClientConnection throws this error. If there is IOException, it is clear that there is network error

File: src/org/loklak/harvester/BaseScraper.java (hunk 2 above)
Comment (vibhcool) [inline src/org/loklak/harvester/BaseScraper.java]: there is a need of argument to be passed in prepareSearchUrl method, if there is no `type` argument, there is need to have an argument, `all` appears to fit here.

File: test/org/loklak/api/search/GithubProfileScraperTest.java (hunk 3 above)
Comment (singhpratyush) [inline test/org/loklak/api/search/GithubProfileScraperTest.java]: You can make the function as `throws IOException` to make network errors fail the build.

Comment (vibhcool) [inline test/org/loklak/api/search/GithubProfileScraperTest.java]: made the suggested changes :)

File: src/org/loklak/api/search/GithubProfileScraper.java [hunk 4]
```diff
@@ -19,35 +19,58 @@
...
 
-    private static final long serialVersionUID = -4166800345379685201L;
+    private final long serialVersionUID = -4166800345379685202L;
     private static final String GITHUB_API_BASE = "https://api.github.com/users/";
+    public List<String> termsList = null;
+
+    public GithubProfileScraper() {
+        super();
+        this.baseUrl = "https://github.com/";
+        String scraperName = "Github";
```
Comment (singhpratyush) [inline src/org/loklak/api/search/GithubProfileScraper.java]: Why are you using this as a local varaible. This should be a class property right?

Comment (vibhcool) [inline src/org/loklak/api/search/GithubProfileScraper.java]: fixed this
